### PR TITLE
Add github actions runner setup for all platforms

### PR DESCRIPTION
**Pull request by anjannath**
this adds the following new flags to the `mapt aws windows create` and `mapt azure windows create` commands:

```
--install-ghactions-runner        Install and setup Github Actions runner in the instance
--ghcations-runner-name string    Name for the Github Actions Runner
--ghcations-runner-repo string    Full URL of the repository where the Github Actions Runner should be registered
--ghcations-runner-token string   Token needed for registering the Github Actions Runner token
```

### CODE REVIEW

**Comment by adrianriobo** [comment]
Fixes #211

File: pkg/provider/azure/action/windows/rhqp-ci-setup.ps1 [hunk 1]
```diff
@@ -124,6 +131,17 @@ if ($crcProfile) {
     New-LocalGroup -Name "crc-users"
     Add-LocalGroupMember -Group "crc-users" -Member $user
 }
+
+# Install github actions runner if requested
+if ($installGHActionsRunner) {
+    New-Item -Path C:\actions-runner -Type Directory ; cd C:\actions-runner
+    Invoke-WebRequest -Uri https://github.com/actions/runner/releases/download/v2.317.0/actions-runner-win-x64-2.317.0.zip -OutFile actions-runner-win-x64-2.317.0.zip
+    Add-Type -AssemblyName System.IO.Compression.FileSystem ;
```
[the remark this answers is not in the document]
Comment (anjannath) [reply pkg/provider/azure/action/windows/rhqp-ci-setup.ps1:139]: okay, i was considering it, then i saw this script didn't currently use templating and instead used args to customize, therefore went with this, i'll update it to use the same snippet

**Comment by adrianriobo** [comment]
Dunno how you will reconcile the commits after re factor but may you can add doc information about the feature.

Like something on the main README.md and also on each mapt usage `doc` inside the `doc` folder

**Comment by adrianriobo** [comment]
Hey can you add the snippet for mac as well? or do you prefer handle it on a different PR?

**Comment by anjannath** [comment]
> Hey can you add the snippet for mac as well? or do you prefer handle it on a different PR?
> 
> Ensure to check [#211 (comment)](https://github.com/redhat-developer/mapt/issues/211#issuecomment-2149063017)

yeah, I'd prefer to do it in a separate PR :)

File: pkg/provider/aws/action/windows/windows.go [hunk 2]
```diff
@@ -42,8 +43,12 @@ type Request struct {
...
 	AMIKeepCopy bool
 	// Features
-	Spot   bool
-	Airgap bool
+	Spot                 bool
+	Airgap               bool
+	SetupGHActionsRunner bool // setup as github actions runner
+	GHActionsRunnerToken string
+	GHActionsRunnerName  string
+	GHActionsRunnerRepo  string
```
[the remark this answers is not in the document]
Comment (anjannath) [reply pkg/provider/aws/action/windows/windows.go]: in the latest push now azure and aws both windows hosts are using the same snippet, but i couldn't figure out how to make the token as a secret variable :(

**Review by adrianriobo — APPROVED** [review]

**Pull request merged**